flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -657,6 +660,117 @@ public void testNotSupportInputSelectableOperatorIfCheckpointing() {
 		StreamingJobGraphGenerator.createJobGraph(env.getStreamGraph());
 	}
 
+	@Test
+	public void testManagedMemoryFractionForSpecifiedResourceSpec() throws Exception {
+		final ResourceSpec resource1 = ResourceSpec.newBuilder(1, 100)
+			.setOnHeapManagedMemory(new MemorySize(100))
+			.setOffHeapManagedMemory(new MemorySize(40))
+			.build();
+		final ResourceSpec resource2 = ResourceSpec.newBuilder(1, 100)
+			.setOnHeapManagedMemory(new MemorySize(300))
+			.setOffHeapManagedMemory(new MemorySize(50))
+			.build();
+		final ResourceSpec resource3 = ResourceSpec.newBuilder(1, 100)
+			.setOnHeapManagedMemory(new MemorySize(600))
+			.setOffHeapManagedMemory(new MemorySize(10))
+			.build();
+		final ResourceSpec resource4 = ResourceSpec.newBuilder(1, 100)
+			.setOnHeapManagedMemory(new MemorySize(123))
+			.setOffHeapManagedMemory(new MemorySize(456))
+			.build();
+
+		// v1, v2 are in the same slot sharing group, v3 is in a different group
```

Comment:
let's add this comment to the other test as well

Reply:
done.